### PR TITLE
Add Repo.config

### PR DESCRIPTION
**Pull request by zshipko**
- Exposes `Repo.config` to access the `Conf.t` that was passed to `Repo.v`

### CODE REVIEW

File: src/irmin/store.ml [hunk 1]
```diff
@@ -233,6 +233,7 @@ module Make (B : Backend.S) = struct
     type t = repo
 
     let v = B.Repo.v
+    let config (t : t) = B.Repo.config t
```
Comment (metanivek) [inline src/irmin/store.ml:236]:
```suggestion
    let config = B.Repo.config
```
Small nit. There is a mixture of styles in this section of code, but I'm partial to this suggestion. If you keep it as-is, is it possible to drop the type specification?

File: src/irmin/backend.ml [hunk 2]
```diff
@@ -94,6 +94,7 @@ module type S = sig
     val contents_t : t -> read Contents.t
     val node_t : t -> read Node.t
     val commit_t : t -> read Commit.t
+    val config : t -> Conf.t
```
Comment (metanivek) [inline src/irmin/backend.ml:97]: I don't particularly love the `_t` suffix but what do you think about `config_t` for consistency?

**Review by metanivek — COMMENTED** [review]
A couple of small suggestions but LGTM

File: src/irmin/backend.ml (hunk 2 above)
Comment (zshipko) [reply src/irmin/backend.ml:97]: Normally I would agree, but in this case the `_t` functions all return backend stores, so I think calling it `config_t` is a bit misleading.

**Review by Ngoguey42 — APPROVED** [review]

**Pull request merged**